gravitee-am-test/specs/gateway/login-flow.jest.spec.ts

```diff
@@ -0,0 +1,388 @@
+        expect(user2TokenResponse.headers['location']).toContain('callback?code=');
+        await logoutUser(openIdConfiguration.end_session_endpoint, user2TokenResponse);
+
+        //user3 has same password as user2
+        const user3TokenResponse = await loginUserNameAndPassword(clientId, user3, commonPassword);
+        expect(user3TokenResponse.headers['location']).toContain('callback?code=');
+        await logoutUser(openIdConfiguration.end_session_endpoint, user3TokenResponse);
+
+        //user3 has same password as user2
+        const user4TokenResponse = await loginUserNameAndPassword(clientId, user4, user4Password);
+        expect(user4TokenResponse.headers['location']).toContain('callback?code=');
+        await logoutUser(openIdConfiguration.end_session_endpoint, user4TokenResponse);
+    });
+
+    if(jdbc === 'jdbc') {
+        console.log("executing jdbc specific test")
+        it("jdbc: both users should be able to login using additional information (email) and password", async () => {
+            const clientId = multiUserLoginApp.settings.oauth.clientId;
+            //log in user3
+            const user1TokenResponse = await loginAdditionalInfoAndPassword(clientId, commonEmail, commonPassword);
+            expect(user1TokenResponse.headers['location']).toContain('callback?code=');
+            await logoutUser(openIdConfiguration.end_session_endpoint, user1TokenResponse);
+
+            //log in user4
+            const user2TokenResponse = await loginAdditionalInfoAndPassword(clientId, commonEmail, user4Password);
+            await logoutUser(openIdConfiguration.end_session_endpoint, user2TokenResponse);
+        });
+
+        it("jdbc: should throw exception when more than one users have same password for login with additional information", async () => {
+            const clientId = multiUserLoginApp.settings.oauth.clientId;
+            const failedLoginResponse = await loginAdditionalInfoAndPassword(clientId, secondCommonEmail, secondCommonPassword);
+            expect(failedLoginResponse.headers['location']).toContain(`error=login_failed&error_code=invalid_user&error_description=Invalid+or+unknown+user`);
+        });
+
+    } else {
+        console.log("executing mongodb specific test")
+        it("mongo: both users should be able to login using additional information (contract) and password", async () => {
+            const clientId = multiUserLoginApp.settings.oauth.clientId;
+
+            //log in user1
+            const user1TokenResponse = await loginAdditionalInfoAndPassword(clientId, contractValue, user1Password);
+            expect(user1TokenResponse.headers['location']).toContain('callback?code=');
+            await logoutUser(openIdConfiguration.end_session_endpoint, user1TokenResponse);
+
+            //log in user4
+            const user2TokenResponse = await loginAdditionalInfoAndPassword(clientId, contractValue, user4Password);
```

Comment:
maybe user2TokenResponse should be named user4TokenResponse if the user4Password is in used?